packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue

```diff
@@ -411,55 +528,104 @@ const onUpdateSelection = (id) => {
 	emit('update-output-port', outputPort);
 };
 
+// Creates a temp config (if doesnt exist in state)
+// This is used for beaker context when there are no outputs in the node
+const createTempModelConfig = async () => {
+	const state = _.cloneDeep(props.node.state);
+	if (state.tempConfigId !== '' || !model.value) return;
+	const data = await createModelConfiguration(
+		model.value.id,
+		'Temp_config_name',
+		'Utilized in model config node for beaker purposes',
+		model.value,
+		true
+	);
+
+	state.initials = knobs.value.initials;
+	state.parameters = knobs.value.parameters;
+	state.timeseries = knobs.value.timeseries;
+	state.tempConfigId = data.id;
+	emit('update-state', state);
+};
+
+// Fill the form with the config data
+const initialize = async () => {
+	const state = _.cloneDeep(props.node.state);
+	const modelId = props.node.inputs[0].value?.[0];
+	if (!modelId) return;
+	model.value = await getModel(modelId);
+
+	knobs.value.name = state.name;
+	knobs.value.description = state.description;
+
+	// State has never been set up:
+	if (state.tempConfigId === '') {
+		// Grab these values from model to inialize them
+		knobs.value.initials =
+			model.value?.semantics?.ode.initials !== undefined
+				? model.value?.semantics?.ode.initials
+				: [];
+		knobs.value.parameters =
+			model.value?.semantics?.ode.parameters !== undefined
+				? model.value?.semantics?.ode.parameters
+				: [];
+		knobs.value.timeseries =
+			model.value?.metadata?.timeseries !== undefined ? model.value?.metadata?.timeseries : [];
+		await createTempModelConfig();
+	}
+	// State already been set up use it instead:
+	else {
+		knobs.value.initials = state.initials;
+		knobs.value.parameters = state.parameters;
+		knobs.value.timeseries = state.timeseries;
+	}
+
+	// Create a new session and context based on model
+	try {
+		const jupyterContext = buildJupyterContext();
+		if (jupyterContext) {
+			await kernelManager.init('beaker_kernel', 'Beaker Kernel', jupyterContext);
+		}
+	} catch (error) {
+		logger.error(`Error initializing Jupyter session: ${error}`);
+	}
+};
+
 watch(
-	() => props.node.state,
-	() => {
-		configInitials.value = props.node.state.initials;
-		configParams.value = props.node.state.parameters;
-		configTimeSeries.value = props.node.state.timeseries;
+	() => props.node.inputs[0],
+	async () => {
+		await initialize();
 	},
-	{ immediate: true, deep: true }
+	{ immediate: true }
+);
```

Comment:
This may be better placed in `onMounted` hook, it is effectively watching the same conditions as we don't really expect inputs to change after the component is created. onMounted gives you a bit more guarantee that the scaffolds are there to use (e.g. the DOM tree).

Reply:
fair point